rwengine/include/script/ScriptTypes.hpp

```diff
@@ -66,13 +68,20 @@ struct SCMOpcodeParameter {
 
 	int integerValue() const 
 	{
-		if ( type == TGlobal )
-		{
-			return * globalInteger;
-		}
-		else
+		switch (type)
 		{
-			return integer;
+		case TGlobal:
+		case TLocal:
+			return *globalInteger;
+		case TInt8:
+			return (int8_t)integer;
```

Comment:
Is this correct? I'm not sure about the internals but I'd assume that `integer` is garbage because it's popping too much data from the stack?
Casting would depend on the endianess then?

(I don't really mind, but we should add a comment about this maybe?)